### PR TITLE
Replace plus char with minus in version tag for GCP dev installation of UCX

### LINKED ISSUE #419: Installation failing for GCP workspace
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/home/bbandi_petsmart_com/ucx/ucx/src/databricks/labs/ucx/install.py", line 713, in <module>
    installer.run()
  File "/home/bbandi_petsmart_com/ucx/ucx/src/databricks/labs/ucx/install.py", line 112, in run
    self._run_configured()
  File "/home/bbandi_petsmart_com/ucx/ucx/src/databricks/labs/ucx/install.py", line 116, in _run_configured
    self._create_jobs()
  File "/home/bbandi_petsmart_com/ucx/ucx/src/databricks/labs/ucx/install.py", line 340, in _create_jobs
    self._deployed_steps[step_name] = self._ws.jobs.create(**settings).job_id
                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/tmp/tmp.UaQueFGpkl/lib/python3.11/site-packages/databricks/sdk/service/jobs.py", line 2999, in create
    res = self._api.do('POST', '/api/2.1/jobs/create', body=body, headers=headers)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/tmp/tmp.UaQueFGpkl/lib/python3.11/site-packages/databricks/sdk/core.py", line 1061, in do
    return retryable(self._perform)(method,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/tmp/tmp.UaQueFGpkl/lib/python3.11/site-packages/databricks/sdk/retries.py", line 47, in wrapper
    raise err
  File "/tmp/tmp.UaQueFGpkl/lib/python3.11/site-packages/databricks/sdk/retries.py", line 29, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/tmp/tmp.UaQueFGpkl/lib/python3.11/site-packages/databricks/sdk/core.py", line 1150, in _perform
    raise self._make_nicer_error(response=response, **payload) from None
databricks.sdk.core.DatabricksError:
The provided tag value (v0.3.1+2320231009205751) contains invalid character. The value must match the regular
expression ^(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?$.

### PR DESCRIPTION
**Pull request by larsgeorge-db**
Fixes #419

### CODE REVIEW

**Review by nfx — CHANGES_REQUESTED** [review]

File: src/databricks/labs/ucx/install.py [hunk 1]
```diff
@@ -607,8 +607,9 @@ def _version(self):
         if hasattr(self, "__version"):
             return self.__version
         project_root = self._find_project_root()
-        if not (project_root / ".git/config").exists():
+        if not (project_root / ".git/config").exists() or self._ws.config.is_gcp():
```
[the remark this answers is not in the document]
Comment (larsgeorge-db) [reply src/databricks/labs/ucx/install.py]: done

**Comment by larsgeorge-db** [comment]
Nope, that does NOT work @nfx 

```
[+] installing dependencies within ephemeral Virtualenv: /var/folders/qf/rb54syb53992k0qhj_y04_yr0000gp/T/tmp.KzZMkpkNuZ
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/Users/lars.george/projects/work/databricks/ucx/src/databricks/labs/ucx/install.py", line 713, in <module>
    installer.run()
  File "/Users/lars.george/projects/work/databricks/ucx/src/databricks/labs/ucx/install.py", line 110, in run
    logger.info(f"Installing UCX v{self._version}")
                                   ^^^^^^^^^^^^^
  File "/Users/lars.george/projects/work/databricks/ucx/src/databricks/labs/ucx/install.py", line 635, in _version
    raise OSError(msg) from None
OSError: Cannot determine unreleased version. Please report this error message that you see on https://github.com/databrickslabs/ucx/issues/new. Meanwhile, download, unpack, and install the latest released version from https://github.com/databrickslabs/ucx/releases. Original error is: Not a valid SemVer: v0.3.1_2420231010124511
```

**Comment by larsgeorge-db** [comment]
We could use minus `-`? https://regex101.com/r/Ly7O1x/2071

File: src/databricks/labs/ucx/install.py [hunk 2]
```diff
@@ -490,9 +490,10 @@ def _job_settings(self, step_name: str, dbfs_path: str):
                 on_success=[self._my_username], on_failure=[self._my_username]
             )
         tasks = sorted([t for t in _TASKS.values() if t.workflow == step_name], key=lambda _: _.name)
+        version = self._version if not self._ws.config.is_gcp else self._version.replace("+", "-")
```
Comment (alexott) [inline src/databricks/labs/ucx/install.py:493]: I would generalize the code to make sure that all tags are cleaned up, not only the version.

**Review by nfx — APPROVED** [review]

**Pull request merged**